### DISCUSSION TITLE
Sessions being merged when selecting a different session

### DISCUSSION THREAD

**Discussion by rbhanot4739** [olimorris/persisted.nvim#164] [Q&A] [answered]
Hello,

First of all thank you for the amazing plugin and it looks very promising. However, I am noticing weird behavior, i.e. when I switch to another session from within an existing session, the previous session gets merged with newly opened session, which is undesirable in my opinion if the sessions are from different directories or projects. Here is step by step breakdown of what I am experiencing

1. Launch nvim in a directory say `dir1` and open a bunch of files, and then quit it. The session is saved correctly; next time you launch vim, the session is restored correctly.
2. Switch to a different git branch or cd to a different project directory (say `dir2`) and repeat step 1.
3. Relaunch nvim; the session is restored correctly with the correct set of buffers being loaded. 
4. Now load the session for `dir2` either with `:Telescope persisted` or `:SessionSelect`,  the buffers from `dir1` and `dir2` are merged together.
5. Quit nvim and cd to `dir1` and launch nvim again; the new loaded session for `dir1` now has buffers from `dir2` merged.

I am not sure if this is the desired behavior, but this sort of breaks the sessions for different directories or branches. The ideal and expected behaviour would be to somehow close all the buffers of the current session before the new session is loaded.

 Btw this is my config
```
{
    "olimorris/persisted.nvim",
    lazy = false, -- make sure the plugin is always loaded at startup
    config = function(_, opts)
      local persisted = require("persisted")
      persisted.branch = function()
        local branch = vim.fn.systemlist("git branch --show-current")[1]
        return vim.v.shell_error == 0 and branch or nil
      end
      persisted.setup(opts)
      require("telescope").setup({
        extensions = {
          persisted = {
            layout_config = { width = 0.55, height = 0.40 },
          },
        },
      })
      require("telescope").load_extension("persisted")
    end,
    opts = {
      autoload = true,
      use_git_branch = true,
      on_autoload_no_session = function()
        vim.notify("No existing session to load.")
      end,
      ignored_dirs = { "~/.config", "~/.local", { "~/", exact = true } },
    },
  }
```

**Comment by olimorris** [comment, marked as answer]
Actually, there's nothing in that behaviour which is unexpected and I touch on your use case in the README, see the **Events/Callbacks** section:

> A commonly requested example is to use the Telescope extension to load a session, saving the current session before clearing all of the open buffers:
> ```lua
> vim.api.nvim_create_autocmd("User", {
>  pattern = "PersistedTelescopeLoadPre",
>  callback = function(session)
>    -- Save the currently loaded session using the global variable
>    require("persisted").save({ session = vim.g.persisted_loaded_session })
>
>    -- Delete all of the open buffers
>    vim.api.nvim_input("<ESC>:%bd!<CR>")
>  end,
> })
> ```

The loading of a session simply uses vim's native source function (`:h :source`) to load the session which results in it being merged into your existing buffers. Because `persisted.start()` has been initiated, when you close Neovim, the `VimLeavePre` autocmd triggers `persisted.save()` which saves all of the buffers into the session.
